fiftyone/core/map/threading.py

```diff
@@ -0,0 +1,240 @@
+"""
+Threading mapping backend
+
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+
+import concurrent.futures
+import queue
+import threading
+from typing import (
+    Any,
+    Callable,
+    Iterator,
+    List,
+    Literal,
+    Optional,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+import bson
+import numpy as np
+from tqdm import tqdm
+
+import fiftyone.core.utils as fou
+import fiftyone.core.view as fov
+import fiftyone.core.map.map as fomm
+import fiftyone.core.map.sequential as foms
+
+
+T = TypeVar("T")
+R = TypeVar("R")
+
+
+class ThreadingMapBackend(foms.SequentialMapBackend):
+    """Executes map_samples with threading using iter_samples."""
+
+    def __init__(self, max_shard_size: Union[int, None] = None):
+        self.__max_shard_size = (
+            max_shard_size
+            or fou.recommend_batch_size_for_value(
+                bson.ObjectId(), max_size=100000
+            )
+        )
+
+    def split_sample_collection(
```

Comment:
I did a small refactor of sample collection splitting to the Batcher class, we can cherry pick over here 

https://github.com/voxel51/fiftyone/blob/c8544d176cd04b98db7c9cdbcc5c9572dfb8484b/fiftyone/core/map/batcher.py

Reply:
I don't want to cherry pick anything. The point of a common branch is to make refactors for common functionality, like I keep talking about in standup.

This function has tests and changes and stuff I don't want to lose.